release_creation/main.py

```diff
@@ -0,0 +1,40 @@
+from enum import Enum
+import os
+import argparse
+
+from github_client.releases import (
+    create_new_release_for_version,
+    get_latest_snapshot_release,
+    add_assets_to_release,
+)
+from snapshot.create import generate_snapshot
+
+BASE_DIR = os.path.dirname(os.path.realpath(__file__))
+
+
+class ReleaseOperations(str, Enum):
```

Comment:
Any reason not to use [`StrEnum`](https://docs.python.org/3/library/enum.html#enum.StrEnum) here instead of (str, Enum)?

Reply:
some issue with needing a new dependency to be backwards compatible with sub 3.11 python but definitely a good find!